src/main/java/org/mafagafogigante/dungeon/stats/ExplorationStatistics.java

```diff
@@ -113,4 +142,21 @@ public int getMaximumNumberOfVisits(Id locationId) {
     return maximumVisitsToLocationWithThisId;
   }
 
+  /**
+   * Returns the number of visits for distinct Locations from a specific part of day.
+   *
+   * @param partOfDay the PartOfDay of when the location was discovered
+   * @return a nonnegative integer
+   */
+  public int getVisitsByPartOfDay(PartOfDay partOfDay) {
+    int count = 0;
+    for (ExplorationStatisticsEntry location : distinctLocations.values()) {
+      if (PartOfDay.getCorrespondingConstant(location.getDiscoveredDate()).getName().getSingular().equalsIgnoreCase(
```

Comment:
Same as above, try to simplify this by adding meaningful local variables.

Reply:
How about this:

```
  public int getVisitsByPartOfDay(PartOfDay partOfDay) {
    int count = 0;
    String locationName;
    for (ExplorationStatisticsEntry location : distinctLocations.values()) {
      locationName = PartOfDay.getCorrespondingConstant(location.getDiscoveredDate()).getName().getSingular();
      if (locationName.equalsIgnoreCase(partOfDay.getName().getSingular())) {
```

Reply:
```java
  public int getVisitsByPartOfDay(PartOfDay partOfDay) {
    int count = 0;
    for (ExplorationStatisticsEntry location : distinctLocations.values()) {
    String locationName = PartOfDay.getCorrespondingConstant(location.getDiscoveredDate()).getName().getSingular();
      if (locationName.equalsIgnoreCase(partOfDay.getName().getSingular())) {
```

would be one line shorter, and `locationName` would never be undefined. If that line gets too long, you can use part of the right-hand side as yet another variable:

```java
  public int getVisitsByPartOfDay(PartOfDay partOfDay) {
    int count = 0;
    for (ExplorationStatisticsEntry location : distinctLocations.values()) {
      <TYPE> <NAME> = PartOfDay.getCorrespondingConstant(location.getDiscoveredDate());
      String locationName = <NAME>.getName().getSingular();
      if (locationName.equalsIgnoreCase(partOfDay.getName().getSingular())) {
```

Tip: you can use ```java to get Java syntax highlighting.

Reply:
Something like this:

```java
 public int getVisitsByPartOfDay(PartOfDay partOfDay) {
    int count = 0;
    for (ExplorationStatisticsEntry location : distinctLocations.values()) {
      PartOfDay day = PartOfDay.getCorrespondingConstant(location.getDiscoveredDate());
      String locationName = day.getName().getSingular();
        if (locationName.equalsIgnoreCase(partOfDay.getName().getSingular())) {
        count++;
      }
    }
    return count;
  }
```

Reply:
Yes, seems right.